index.js

```diff
@@ -11,30 +10,54 @@ module.exports = {
   init (ssb, config) {
     const allowedTypes = getAllowedTypes(ssb, config)
 
-    ssb.publish.hook((publish, args) => {
+    const publishHook = ({ db2 }) => (publish, args) => {
```

Comment:
Split this into two functions IMO:
- `publishHook`
- `createHook`

smaller functions, less nesting, less ternaries, more clarity

Reply:
:fire: Yeah actually this is a strong opinion - this would be much easier to review if I wasn't having to juggle the context of db1 / db2 at each step of this function. Please change  (context I'm tired from night wake ups with a kid, have less working memory today)

Reply:
i'm not sure i can write 2 bugfree functions that are this complex but i'll try and see how it ends up

Reply:
i split them now